pkg/driver/driver.go

```diff
@@ -131,3 +135,65 @@ func Boot(ctx context.Context, d Driver, pw progress.Writer) (*client.Client, st
 		return c, chosenNodeName, nil
 	}
 }
+
+// GetPlatforms returns a de-duped set of available platforms across all nodes
+// and a bool to indicate a "mixed" cluster with nodes of different types. False
+// indicates a non-mixed cluster.  If the some nodes in the cluster contain a
+// subset of supported platforms of others in the cluster (e.g. newer
+// generations of the same family) False will be returned.
+// TODO - bool to skip offline nodes?
+func (info *Info) GetPlatforms() ([]specs.Platform, bool) {
+	isMixed := false
+	latestPlatforms := []specs.Platform{}
+	allPlatforms := []specs.Platform{}
+	for _, node := range info.DynamicNodes {
+		if len(node.Platforms) == 0 {
+			continue
+		}
+		allPlatforms = append(allPlatforms, node.Platforms...)
+		if isMixed {
```

Comment:
not quite sure what's going on here. if isMixed got set, wouldn't this just iterate really fast? would it make more sense to just break out of the loop?

That kinda seems wrong though. Would it just be easier to make a hashmap of the platforms, iterate through each of them and then return a list of the keys in the hashmap?